code/game/machinery/vending.dm

```diff
@@ -357,32 +356,39 @@
 							O.forceMove(loc)
 					new /obj/item/electronics/circuitboard/vending(loc)
 					qdel(src)
-
 		if(ABORT_CHECK)
 			return
 
 	var/obj/item/card/id/ID = I.GetIdCard()
 
 	if(currently_vending && earnings_account && !earnings_account.suspended)
-		var/paid = 0
-		var/handled = 0
-
-		if(ID) //for IDs and PDAs and wallets with IDs
-			paid = pay_with_card(ID,I)
-			handled = 1
-			playsound(usr.loc, 'sound/machines/id_swipe.ogg', 100, 1)
-		else if(istype(I, /obj/item/spacecash/ewallet))
-			var/obj/item/spacecash/ewallet/C = I
-			paid = pay_with_ewallet(C)
-			handled = 1
-			playsound(usr.loc, 'sound/machines/id_swipe.ogg', 100, 1)
-		else if(istype(I, /obj/item/spacecash/bundle))
-			var/obj/item/spacecash/bundle/C = I
-			paid = pay_with_cash(C)
-			handled = 1
+		var/paid = FALSE
+		var/handled = FALSE
+
+		if(alt_currency_path)
+			if(istype(I, alt_currency_path))
+				paid = pay_with_item(I, user)
+			else
+				var/atom/movable/AM = alt_currency_path
+				to_chat(user, SPAN_WARNING("This vending machine only accepts [initial(AM.name)] as currency."))
```

Comment:
Can you take a var from an un-instanced path, i thought you could only do it by using initial?